### ISSUE TITLE
ChatMistralAI doesn't work for models without tools support [422]

### ISSUE THREAD

**Issue by OlegIvaniv** [langchain-ai/langchainjs#4646]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain.js documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain.js rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

```typescript
import { ChatMistralAI } from "@langchain/mistralai";
import { ChatPromptTemplate } from "@langchain/core/prompts";

const model = new ChatMistralAI({
  apiKey: process.env.MISTRAL_API_KEY,
  modelName: "mistral-tiny",
});
const prompt = ChatPromptTemplate.fromMessages([
  ["system", "You are a helpful assistant"],
  ["human", "{input}"],
]);
const chain = prompt.pipe(model);
const response = await chain.invoke({
  input: "Hello",
});
console.log("response", response);
/**
```

### Error Message and Stack Trace (if applicable)

Request failed: HTTP error! status: 422 Response: 
```json
{
  "object": "error",
  "message": {
    "detail": [
      {
        "type": "extra_forbidden",
        "loc": [
          "body",
          "tools"
        ],
        "msg": "Extra inputs are not permitted",
        "input": [],
        "url": "https://errors.pydantic.dev/2.5/v/extra_forbidden"
      }
    ]
  },
  "type": "invalid_request_error",
  "param": null,
  "code": null
}
```

### Description

When using `ChatMistralAI` with a model that doesn't have support for tools calling, the Mistral API returns 422. This is because we always populate the `tools` property when sending the request. So effectively the only working models are `mistral-small-latest`, `mistral-small-2402`, `mistral-large`,.

### System Info

❯ pnpm info langchain

langchain@0.1.25 | MIT | deps: 18 | versions: 255
Typescript bindings for langchain
https://github.com/langchain-ai/langchainjs/tree/main/langchain/

keywords: llm, ai, gpt3, chain, prompt, prompt engineering, chatgpt, machine learning, ml, openai, embeddings, vectorstores

dist
.tarball: https://registry.npmjs.org/langchain/-/langchain-0.1.25.tgz
.shasum: afd0faf2eadfb5e6e1d57bb3bbc5731918e62135
.integrity: sha512-sfEChvr4H2CklHdSByNBbytwBrFhgtA5kPOnwcBrxuXGg1iOaTzhVxQA0QcNcQucI3hZrsNbZjxGp+Can1ooZQ==
.unpackedSize: 4.0 MB

dependencies:
@anthropic-ai/sdk: ^0.9.1     js-tiktoken: ^1.0.7           openapi-types: ^12.1.3        
@langchain/community: ~0.0.33 js-yaml: ^4.1.0               p-retry: 4                    
@langchain/core: ~0.1.36      jsonpointer: ^5.0.1           uuid: ^9.0.0                  
@langchain/openai: ~0.0.14    langchainhub: ~0.0.8          yaml: ^2.2.1                  
binary-extensions: ^2.2.0     langsmith: ~0.1.7             zod-to-json-schema: ^3.22.3   
expr-eval: ^2.0.2             ml-distance: ^4.0.0           zod: ^3.22.4                  

maintainers:
- nfcampos <nuno@boringbits.io>
- jacoblee93 <jacoblee93@gmail.com>
- davidduong <david@duong.cz>
- hwchase17 <hw.chase.17@gmail.com>
- basproul <braceasproul@gmail.com>

dist-tags:
latest: 0.1.25     next: 0.1.19-rc.2  

published 5 days ago by basproul <braceasproul@gmail.com>

**Comment by OlegIvaniv** [comment]
Perhaps not having the `?? []` fallback here would be enough. That would only send the tools if there are any and it would be up-to the consumer of the model to decide. 
```ts
  invocationParams(
    options?: this["ParsedCallOptions"]
  ): Omit<MistralAIChatCompletionOptions, "messages"> {
    const { response_format, tools, tool_choice } = options ?? {};
    const mistralAITools: MistralAIToolInput[] =
      tools
        ?.map((tool) => {
          if ("lc_namespace" in tool) {
            return _convertStructuredToolToMistralTool([tool]);
          }
          return tool;
        })
        .flat() ?? [];
```
@bracesproul Since you've worked on this part of the codebase recently, do you have any thoughts?

Cheers

**Comment by jacoblee93** [comment]
Thanks for the report! Will push a fix shortly.

**Comment by jacoblee93** [comment]
Will have a look, thanks for flagging. What is `"modelName": this.configuration.modelId,` here?

**Comment by jacoblee93** [comment]
Got it, will report back.

**Comment by PeterAronZentai** [comment]
But this must somehow be related to the service / platform, like an extra json validation step been added  - we have a system untouched in terms of deployments working fine for weeks, and mistral small models broke just in past days.

**Comment by jacoblee93** [comment]
Apologies for the regression - I've patched this in #5139 and will release a new version (0.0.19) shortly.

**Comment by jacoblee93** [comment]
Thank you for the report! Seems there actually was a backend Mistral change as well.

Please ping again if you see something similar recur.
